### ISSUE TITLE
Folder name specific colors

### ISSUE THREAD

**Issue by antoniosarosi** [nvim-tree/nvim-tree.lua#2369]
**Is this a question?**
Original Q&A: https://github.com/nvim-tree/nvim-tree.lua/discussions/1498

**Can this functionality be implemented utilising API?**
I don't think so, I haven't found any function to change the color of an icon.

**Is your feature request related to a problem? Please describe.**
It's not related to a problem or bug, the feature would simply allow the user to change the color of a directory icon based on its name, for example, `"node_modules" = "green"`, `"lua" = "blue"`, etc.

**Describe the solution you'd like**
Either a setup option similar to `nvim-web-devicons`:

```lua
require'nvim-web-devicons'.setup {
 override_by_filename = {
  [".gitignore"] = {
    icon = "",
    color = "#f1502f",
    name = "Gitignore"
  }
 };
}
```

Maybe something like this:

```lua
require("nvim-tree").setup({
    renderer = {
        indent_markers = {
            enable = true,
        },
        override_by_directory_name = {
            ["node_modules"] = {
                icon = "󰎙",
                color = "#68A063",
                name = "NodeModules",         
            }
        }
    }
})
```

Or some sort of API function that gets called with the directory name and returns the color (not sure how that would work or how it would impact performance).

**Describe alternatives you've considered**
I tried configuring `nvim-web-devicons` directly but I think you can only to that with files, not directories.

**Additional context**
Basically trying to simulate icon themes like `Material Icon Theme` for VScode:

![](https://raw.githubusercontent.com/PKief/vscode-material-icon-theme/main/images/folderIcons.png)

**Comment by alex-courtis** [comment]
This can be done. We can match the folder based on nvim-web-devicons name.

Something like `M.devicons.get_icon(node.name, node.extension)` in `icons.lua get_folder_icon`

The icon and colour would be provided from overrides in nvim-web-devicons

[Pull requests](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md) are always gratefully appreciated.

**Comment by alex-courtis** [comment]
My initial notes around highlight groups were for folder colour, however as that is not needed, this much simpler solution is appropriate.

**Comment by antoniosarosi** [comment]
You mean [here](https://github.com/nvim-tree/nvim-tree.lua/blob/master/lua/nvim-tree/renderer/components/icons.lua#L12)? I'll try to implement it myself and submit a PR. I don't know much about Lua but seems to be a simple language.

**Comment by alex-courtis** [comment]
> You mean [here](https://github.com/nvim-tree/nvim-tree.lua/blob/master/lua/nvim-tree/renderer/components/icons.lua#L12)? I'll try to implement it myself and submit a PR.

That would be grand. Please add a default off option to enable this behaviour: `renderer.icons.webdev_colors_folder` to control this behaviour.

> I don't know much about Lua but seems to be a simple language.

Famous last words.
